codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketServerHandshaker.java

```diff
@@ -140,12 +147,33 @@ public int maxFramePayloadLength() {
      *              HTTP Request
      * @return future
      *              The {@link ChannelFuture} which is notified once the opening handshake completes
+     * @deprecated use {@link #handshake(Channel, FullHttpRequest, Set)}
      */
+    @Deprecated
     public ChannelFuture handshake(Channel channel, FullHttpRequest req) {
         return handshake(channel, req, null, channel.newPromise());
     }
 
     /**
+     * Performs the opening handshake. When call this method you <strong>MUST NOT</strong> retain the
+     * {@link FullHttpRequest} which is passed in.
+     *
+     * @param channel
+     *              Channel
+     * @param req
+     *              HTTP Request
```

Comment:
I'm not sure `channel` and `req` parameter descriptions are very useful. They seem redundant to me. I think there's no need to add `@param` for all parameters unless you have something to say really.